### PR TITLE
K8SPSMDB-329: improve object updates

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPSMDB-329](https://jira.percona.com/browse/K8SPSMDB-329) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| init-deploy | passed |
| one-pod | failed |
| upgrade-consistency | failed |
| demand-backup | failed |
| self-healing | passed |
| limits | failed |
| monitoring-2-0 | failed |
| scaling | passed |
| scheduled-backup | failed |
| security-context | failed |
| operator-self-healing | passed |
| demand-backup-sharded | failed |
| arbiter | failed |
| rs-shard-migration | failed |
| service-per-pod | failed |
| upgrade | passed |
| liveness | failed |
| upgrade-sharded | failed |
| smart-update | failed |
| version-service | failed |
| pitr | failed |
| pitr-sharded | failed |
| users | passed |
| data-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/ba379cd29de29a58cb339efb18f3f004d3d6b13f
image: `perconalab/percona-server-mongodb-operator:PR-670-ba379cd2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| rs-shard-migration | failed |
| liveness | passed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | failed |
| users | passed |
| data-sharded | failed |
| pitr | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/4e93e505def25e0c7e1bc303ba975bde52c26ee6
image: `perconalab/percona-server-mongodb-operator:PR-670-4e93e505`

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | failed |
| arbiter | passed |
| service-per-pod | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | failed |
| liveness | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/5552642a8e6f668a9722a9b38b214ebdc0edbcf3
image: `perconalab/percona-server-mongodb-operator:PR-670-5552642a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| init-deploy | passed |
| limits | passed |
| data-sharded | failed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | failed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/4e93e505def25e0c7e1bc303ba975bde52c26ee6
image: `perconalab/percona-server-mongodb-operator:PR-670-4e93e505`

**Review by delgod — DISMISSED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
@qJkee  'some-name-mongos' service  is constantly  recreated
```
some-name-mongos   ClusterIP   10.7.242.40   <none>        27017/TCP   4s
some-name-mongos   ClusterIP   10.7.242.40   <none>        27017/TCP   8s
some-name-mongos   ClusterIP   10.7.243.137   <none>        27017/TCP   0s
some-name-mongos   ClusterIP   10.7.243.137   <none>        27017/TCP   9s
some-name-mongos   ClusterIP   10.7.242.116   <none>        27017/TCP   0s
some-name-mongos   ClusterIP   10.7.242.116   <none>        27017/TCP   9s
some-name-mongos   ClusterIP   10.7.252.244   <none>        27017/TCP   0s
some-name-mongos   ClusterIP   10.7.252.244   <none>        27017/TCP   8s
some-name-mongos   ClusterIP   10.7.246.253   <none>        27017/TCP   0s
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| pitr | passed |
| users | passed |
| pitr-sharded | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/f91234516d59f0df6ed96d7e12d09ee18b5fdcff
image: `perconalab/percona-server-mongodb-operator:PR-670-f9123451`

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 1]
```diff
@@ -933,10 +929,18 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongos(cr *api.PerconaServerMon
 	}
 
 	msDepl.Spec = deplSpec
-	err = r.createOrUpdate(msDepl, msDepl.Name, msDepl.Namespace)
-	if err != nil {
-		return errors.Wrapf(err, "update or create deployment %s", msDepl.Name)
+	if cr.CompareVersion("1.8.0") >= 0 {
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/psmdb_controller.go:932]: `1.8.0` or `1.9.0` ?

Comment (delgod) [reply pkg/controller/perconaservermongodb/psmdb_controller.go:932]: `1.9.0`

Comment (qJkee) [reply pkg/controller/perconaservermongodb/psmdb_controller.go]: fixed in 757afff

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | failed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/757afffd79b54c5327acf71994306b70302d567b
image: `perconalab/percona-server-mongodb-operator:PR-670-757afffd`

**Review by delgod — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| storage | passed |
| version-service | passed |
| self-healing | passed |
| users | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/670/commits/757afffd79b54c5327acf71994306b70302d567b
image: `perconalab/percona-server-mongodb-operator:PR-670-757afffd`

**Pull request merged**